kubetest/local.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"time"
+)
+
+var (
+	localUpTimeout = flag.Duration("local-up-timeout", 2*time.Minute, "(local only) Time limit between 'local-up-cluster.sh' and a response from the Kubernetes API.")
+)
+
+type localCluster struct {
+	tempDir    string
+	kubeConfig string
+}
+
+var _ deployer = localCluster{}
+
+func newLocalCluster() *localCluster {
+	tempDir, err := ioutil.TempDir("", "kubetest-local")
+	if err != nil {
+		log.Fatal("unable to create temp directory")
+	}
+	os.Chmod(tempDir, 0755)
+	return &localCluster{
+		tempDir:    tempDir,
+		kubeConfig: "/var/run/kubernetes/admin.kubeconfig",
```

Comment:
do you need to check permission there?

Reply:
it's created later. no need to check here

Reply:
will do that in another section below